addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -1174,6 +1177,39 @@ export class DeletePlugin extends Plugin {
         }
     }
 
+    /**
+     * Beforeinput event of type deleteContentBackward cannot be default
+     * prevented in Android Chrome. So we need to revert:
+     * - eventual mutations between beforeinput and input events
+     * - eventual selection change after input event
+     *
+     * @param {InputEvent} beforeInputEvent
+     */
+    preventDefaultDeleteAndroidChrome(beforeInputEvent) {
+        const restoreDOM = this.dependencies.history.makeSavePoint();
+        this.onInput = (ev) => {
+            if (ev.inputType !== beforeInputEvent.inputType) {
+                return;
+            }
+            // Revert DOM changes that occurred between beforeinput and input.
+            restoreDOM();
+
+            // Revert selection changes after input event, within the same tick.
+            // If further mutations occurred, consider selection change legit
+            // (e.g. dictionary input) and do not revert it.
+            const { restore: restoreSelection } = this.dependencies.selection.preserveSelection();
+            const observerOptions = { childList: true, subtree: true, characterData: true };
+            const getMutationRecords = observeMutations(this.editable, observerOptions);
+            this.onSelectionChange = () => {
+                const shouldRevertSelectionChanges = !getMutationRecords().length;
+                if (shouldRevertSelectionChanges) {
+                    restoreSelection();
+                }
+            };
+            setTimeout(() => delete this.onSelectionChange);
+        };
```

Comment:
On one hand, I don't like to redefine methods on `this` on the fly like that. I think these should be actual methods that call `this.restoreAndroidChromeDOM` and look for `this.androidChromeMutationRecords`.

On the other hand, an argument I can find in favor of this way of doing things is that all the stuff related to this android chrome hack is self-contained in this method. That's a valuable bonus. In fact I think I would push this even more and rename `this.onInput` to `this.onAndroidChromeInput`, `this.selectionChange` to `this.onAndroidChromeSelectionChange`, and add an actual test that we're really in Android Chrome, because otherwise we're doing this on Firefox Desktop too.

Reply:
I have renamed the methods. But kept them redefined on the fly.
As a reminder, the actual input and selection change handlers are defined as anonymous functions directly in the resources declarations, and they call these androidChrome specific methods if they exist.